File: C_repo/test_10.c
#include <stdio.h>
#include <stdlib.h>

// Tests 'int**' (pointer to a pointer), often used for
// "out parameters" where a function allocates memory for a caller.
// Java would simulate this by having the function *return* the object.

void createInt(int** p) {
    *p = (int*)malloc(sizeof(int));
    **p = 42;
}

int main() {
    int* myIntPtr = NULL;
    createInt(&myIntPtr);
    
    if (myIntPtr != NULL) {
        printf("Value from out-param: %d\n", *myIntPtr);
        free(myIntPtr);
    }
    return 0;
}

File: Runs/Java_repo_codegemma:7b/Test_10.java
public class Test_10 {

    public static void createInt(int[] p) {
            p = new int[1];
            p[0] = 42;
        }

    public static void main(String[] args) {
        int myIntPtr = 0;
        createInt(myIntPtr);
        if (myIntPtr != 0) {
          System.out.printf("Value from out-param: %d\n", myIntPtr);
          free(myIntPtr);
        }
      }

}